perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,400 @@
+
+        params = {
+            self.PCHANNEL_NAME: channel,
+        }
+
+        path = urijoin(self.base_url, self.RCHANNEL_INFO)
+        response = self.fetch(path, params)
+
+        return response
+
+    def messages(self, channel, from_date, offset):
+        """Fetch messages from a channel."""
+
+        query = '{"_updatedAt": {"$gte": {"$date": "%s"}}}' % from_date.isoformat()
+
+        params = {
+            "roomName": channel,
+            "sort": '{"_updatedAt": 1}',
+            "count": self.max_items,
+            "offset": offset,
+            "query": query
+        }
+
+        path = urijoin(self.base_url, self.RCHANNEL_MESSAGES)
+        response = self.fetch(path, params)
+
+        return response
+
+    def fetch(self, url, payload=None, headers=None):
+        """Fetch the data from a given URL.
+
+        :param url: link to the resource
+        :param payload: payload of the request
+        :param headers: headers of the request
+
+        :returns a response object
+        """
+        headers = {
+            self.HAUTH_TOKEN: self.api_token,
+            self.HUSER_ID: self.user_id
+        }
+
+        logger.debug("RocketChat client message request with params: %s", str(payload))
+
+        if not self.from_archive:
+            self.sleep_for_rate_limit()
+
+        response = super().fetch(url, payload, headers=headers)
+
+        if not self.from_archive:
+            self.update_rate_limit(response)
+
+        return response.text
+
+    @staticmethod
+    def sanitize_for_archive(url, headers, payload):
+        """Sanitize payload of a HTTP request by removing the token and
+         user id information before storing/retrieving archived items.
+
+        :param: url: HTTP url request
+        :param: headers: HTTP headers request
+        :param: payload: HTTP payload request
+
+        :returns: url, headers and the sanitized payload
+        """
+        if RocketChatClient.HAUTH_TOKEN in headers:
+            headers.pop(RocketChatClient.HAUTH_TOKEN)
+
+        if RocketChatClient.HUSER_ID in headers:
+            headers.pop(RocketChatClient.HUSER_ID)
+
+        return url, headers, payload
+
+
+class RocketChatCommand(BackendCommand):
+    """Class to run RocketChat backend from the command line."""
+
+    BACKEND = RocketChat
+
+    @classmethod
+    def setup_cmd_parser(cls):
+        """Returns the RocketChat argument parser."""
+
+        parser = BackendCommandArgumentParser(cls.BACKEND,
+                                              from_date=True,
+                                              token_auth=True,
+                                              archive=True,
+                                              ssl_verify=True)
+
+        # Backend token is required
+        action = parser.parser._option_string_actions['--api-token']
+        action.required = True
+
+        # Required argument
+        parser.parser.add_argument('-u', '--user_id',
```

Comment:
When running `perceval rocketchat --help`, the user_id appears as an optional argument (see below), however it doesn't seem possible to fetch messages without providing an user_id.

```
[2020-05-03 11:06:14,424] - Sir Perceval is on his quest.
usage: perceval [-h] [--category CATEGORY] [--tag TAG] [--filter-classified]
                [--from-date FROM_DATE] -t API_TOKEN
                [--archive-path ARCHIVE_PATH] [--no-archive] [--fetch-archive]
                [--archived-since ARCHIVED_SINCE] [--no-ssl-verify]
                [-o OUTFILE] [--json-line] [-u USER_ID]
                [--max-items MAX_ITEMS] [--sleep-for-rate]
                [--min-rate-to-sleep MIN_RATE_TO_SLEEP]
                url channel

positional arguments:
  url                   URL of the RocketChat server
  channel               RocketChat channel(room) name

optional arguments:
  -h, --help            show this help message and exit
  -u USER_ID, --user_id USER_ID
                        User Id to fetch messages

...

```
If line 377 is changed to the code below, the user_id appears as a positional argument

```
parser.parser.add_argument('user_id',
                                   help="User Id to fetch messages")
```